bgpd/bgp_route.c

```diff
@@ -4438,8 +4438,12 @@ int bgp_nlri_parse_ip(struct peer *peer, struct attr *attr,
 					   BGP_ROUTE_NORMAL, NULL, NULL, 0,
 					   NULL);
 
-		/* Address family configuration mismatch or maximum-prefix count
-		   overflow. */
+		/* Do not send BGP notification twice when maximum-prefix count
+		 * overflow. */
+		if (CHECK_FLAG(peer->sflags, PEER_STATUS_PREFIX_OVERFLOW))
```

Comment:
The first notification is sent from bgp_maximum_prefix_overflow() function and 2nd notification will be sent from the caller of this function. So this code modifications will avoid sending two notifications. However, I have a question. 

1. In the old code, when we hit this situation, the functions returns right away. i.e. it will exit from the for loop and stops any further processing. With the new code, even after we hit max_prefix limit, it will still continue to process the NLRI. Do we still need to process this update message when we hit this situation ?

Reply:
I got your point. `break` cannot be used here, because it will return in the middle of processing. Are you suggesting to move this check here https://github.com/FRRouting/frr/blob/master/bgpd/bgp_packet.c#L1574?

Reply:
Yes, that would be one way of doing it. 

or bgp_nlri_parse_ip() function can return an error code instead of just "-1" depending on the error it encounters and then calling function will send the notification accordingly.

Reply:
@srimohans I changed with error codes. Could you take a look again?